l10n_it_fatturapa_out_rc/models/account_move.py

```diff
@@ -81,42 +81,42 @@ def preventive_checks(self):
                     "select invoices with same supplier"
                 )
             )
+        rc_supplier = rc_suppliers and rc_suppliers[0] or rc_suppliers.browse()
+        # --- preventive checks related to set CedentePrestatore.Sede --- #
+        if not rc_supplier.street:
+            raise UserError(
+                _("Partner %s, Street is not set.") % rc_supplier.display_name
+            )
+        if not rc_supplier.city:
+            raise UserError(
+                _("Partner %s, City is not set.") % rc_supplier.display_name
+            )
+        if not rc_supplier.country_id:
+            raise UserError(
+                _("Partner %s, Country is not set.") % rc_supplier.display_name
+            )
+        if not rc_supplier.zip:
+            raise UserError(_("Partner %s, ZIP is not set.") % rc_supplier.display_name)
         # --- preventive checks related to set CedentePrestatore.DatiAnagrafici --- #
-        partner = rc_suppliers and rc_suppliers[0] or self.env["res.partner"].browse()
         fiscal_document_type_codes = invoices_with_rc.mapped(
             "fiscal_document_type_id.code"
         )
-        # Se vale IT , il sistema verifica che il TipoDocumento sia diverso da
-        # TD17, TD18 e TD19; in caso contrario il file viene scartato
-        vat = partner.vat and partner.vat[0:2]
-        if vat:
-            if vat == "IT" and any(
-                [x in ["TD17", "TD18", "TD19"] for x in fiscal_document_type_codes]
-            ):
-                raise UserError(
-                    _(
-                        "A self-invoice cannot be issued with IT country code and "
-                        "fiscal document type in 'TD17', 'TD18', 'TD19'."
-                    )
+        # TD17, TD18 e TD19 solo per partner esteri; in caso contrario il file viene scartato
+        ccode = rc_supplier.country_id.code
+        if ccode == "IT" and any(
+            [x in ["TD17", "TD18", "TD19"] for x in fiscal_document_type_codes]
+        ):
+            raise UserError(
+                _(
+                    "A self-invoice cannot be issued with IT country code and "
+                    "fiscal document type in 'TD17', 'TD18', 'TD19'."
                 )
-            if vat not in self.env["res.country"].search([]).mapped("code"):
-                raise ValueError(
-                    _(
-                        "Country code does not exist or it is not mapped in countries: "
-                        "%s" % vat
-                    )
+            )
+        if ccode not in self.env["res.country"].search([]).mapped("code"):
+            raise ValueError(
+                _(
+                    "Country code does not exist or it is not mapped in countries: "
+                    "%s" % ccode
```

Comment:
Questa sostituzione andrebbe fuori da `_(...)`; ok che era così anche prima ma questo farebbe mostrare l'errore in inglese e non è una grossa correzione da fare, riesci mica a correggerlo?

Reply:
Fatto.